File: 424-longest-repeating-character-replacement/longest-repeating-character-replacement.java
import java.util.HashMap;

class Solution {
    public int characterReplacement(String s, int k) {
        HashMap<Character, Integer> m = new HashMap<Character, Integer>();
        int maxf = 0;
        int result = 0;
        int l = 0;
        for(int r = 0; r < s.length(); r++){
            if(m.containsKey(s.charAt(r))) m.put(s.charAt(r), m.get(s.charAt(r)) + 1);
            else m.put(s.charAt(r), 1);
            maxf = Math.max(maxf, m.get(s.charAt(r)));
            if((r-l+1) - maxf > k){
                m.put(s.charAt(l), m.get(s.charAt(l)) - 1);
                l += 1;
            }
            result = Math.max(result, r-l+1);
        }
        return result; 
    }
}

File: 424-longest-repeating-character-replacement/longest-repeating-character-replacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        dictionary = {}
        mostFrequent = 0
        l = 0
        result = 0
        for r in range(len(s)):
            dictionary[s[r]] = 1 + dictionary.get(s[r], 0)
            mostFrequent = max(mostFrequent, dictionary[s[r]])
            while (r - l + 1) - mostFrequent > k:
                dictionary[s[l]] -=1
                l += 1
            print(l, r)
            result = max(result, r - l + 1)
        print(dictionary, r, l)
        return result
